src/main/scala/temple/generate/server/go/service/GoServiceDAOGenerator.scala

```diff
@@ -191,6 +192,148 @@ object GoServiceDAOGenerator {
       when(operations.contains(Delete)) { GoCommonDAOGenerator.generateExecuteQuery() },
     )
 
+  private def generateInterfaceFunctionComment(
+    serviceName: String,
+    operation: CRUD,
+    createdByAttribute: CreatedByAttribute,
+  ): String =
+    mkCode(
+      "//",
+      generateDAOFunctionName(operation, serviceName),
+      operation match {
+        case List   => s"returns a list containing every $serviceName"
+        case Create => s"creates a new $serviceName"
+        case Read   => s"returns the $serviceName"
+        case Update => s"updates the $serviceName"
+        case Delete => s"deletes the $serviceName"
+      },
+      "in the datastore",
+      operation match {
+        case List =>
+          createdByAttribute match {
+            case _: EnumerateByCreator => "for a given ID"
+            case _                     => ""
+          }
+        case Create => s", returning the newly created $serviceName"
+        case Read   => "for a given ID"
+        case Update => s"for a given ID, returning the newly updated $serviceName"
+        case Delete => "for a given ID"
+      },
+    )
+
+  private def generateListInterfaceFunctionBodyQueryBlock(
+    createdByAttribute: CreatedByAttribute,
+    query: String,
+  ): String =
+    mkCode.lines(
+      CodeWrap.parens.prefix("rows, err := executeQueryWithRowResponses")(
+        mkCode.list(
```

Comment:
I'm sure we've spoken about this before, but it really feels like we're hitting the point of making some Go utility functions that abstract away the mkCode logic a bit

```scala
callGoFunction(returnVars = "rows, err", function = "executeQueryWithRowResponses", args)
```

Reply:
Think I'll look into this in a new PR, where I refactor GoServiceDAOGenerator into separate files too.